crates/astria-core/src/sequencer/v1alpha1/transaction/action.rs

```diff
@@ -975,3 +1012,198 @@ enum FeeAssetChangeActionErrorKind {
+    // the address on the destination chain to send the transfer to.
+    pub destination_chain_address: String,
+}
+
+impl BridgeLockAction {
+    #[must_use]
+    pub fn into_raw(self) -> raw::BridgeLockAction {
+        raw::BridgeLockAction {
+            to: self.to.to_vec(),
+            amount: Some(self.amount.into()),
+            asset_id: self.asset_id.as_bytes().to_vec(),
+            fee_asset_id: self.fee_asset_id.as_ref().to_vec(),
+            destination_chain_address: self.destination_chain_address,
+        }
+    }
+
+    #[must_use]
+    pub fn to_raw(&self) -> raw::BridgeLockAction {
+        raw::BridgeLockAction {
+            to: self.to.to_vec(),
+            amount: Some(self.amount.into()),
+            asset_id: self.asset_id.as_bytes().to_vec(),
+            fee_asset_id: self.fee_asset_id.as_ref().to_vec(),
+            destination_chain_address: self.destination_chain_address.clone(),
+        }
+    }
+
+    /// Convert from a raw, unchecked protobuf [`raw::BridgeLockAction`].
+    ///
+    /// # Errors
+    ///
+    /// - if the `to` field is invalid
+    /// - if the `asset_id` field is invalid
+    /// - if the `fee_asset_id` field is invalid
+    pub fn try_from_raw(proto: raw::BridgeLockAction) -> Result<Self, BridgeLockActionError> {
+        let to =
+            Address::try_from_slice(&proto.to).map_err(BridgeLockActionError::invalid_address)?;
+        let amount = proto
+            .amount
+            .ok_or(BridgeLockActionError::missing_amount())?;
+        let asset_id = asset::Id::try_from_slice(&proto.asset_id)
+            .map_err(BridgeLockActionError::invalid_asset_id)?;
+        let fee_asset_id = asset::Id::try_from_slice(&proto.fee_asset_id)
+            .map_err(BridgeLockActionError::invalid_fee_asset_id)?;
+        Ok(Self {
+            to,
+            amount: amount.into(),
+            asset_id,
+            fee_asset_id,
+            destination_chain_address: proto.destination_chain_address,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct BridgeLockActionError(BridgeLockActionErrorKind);
+
+impl BridgeLockActionError {
+    #[must_use]
+    fn invalid_address(err: IncorrectAddressLength) -> Self {
+        Self(BridgeLockActionErrorKind::InvalidAddress(err))
+    }
+
+    #[must_use]
+    fn missing_amount() -> Self {
+        Self(BridgeLockActionErrorKind::MissingAmount)
+    }
+
+    #[must_use]
+    fn invalid_asset_id(err: asset::IncorrectAssetIdLength) -> Self {
+        Self(BridgeLockActionErrorKind::InvalidAssetId(err))
+    }
+
+    #[must_use]
+    fn invalid_fee_asset_id(err: asset::IncorrectAssetIdLength) -> Self {
+        Self(BridgeLockActionErrorKind::InvalidFeeAssetId(err))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum BridgeLockActionErrorKind {
+    #[error("the address was invalid")]
+    InvalidAddress(#[source] IncorrectAddressLength),
+    #[error("the amount was missing")]
```

Comment:
```suggestion
    #[error("the `amount` field was not set")]
```
I think this makes it a bit more clear what the issue is.